### PR TITLE
IAM: Add TestPermissions code snippets and tests

### PR DESCRIPTION
**Pull request by melaniedejong**
Added TestPermissions code snippets and tests for this doc: https://cloud.google.com/iam/docs/testing-permissions

Also cleaned up access tests by adding helper functions

### CODE REVIEW

**Comment by melaniedejong** [comment]
Apparently the test is failing because the dotnet-docs-samples-tests has exceeded the number of allowed roles (300)--meaning, I can't create a custom role for the test. Is this specific to the AccessTest, or is this a global problem?

**Comment by melaniedejong** [comment]
The tests are passing for me locally now. Also hopefully the linting I did will fix the presubmit error.

File: iam/api/Access/TestPermissions.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// [START iam_test_permissions]
+
+using System;
+using System.Collections.Generic;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.CloudResourceManager.v1;
+using Google.Apis.CloudResourceManager.v1.Data;
+
+public partial class AccessManager
```
Comment (simonz130) [inline iam/api/Access/TestPermissions.cs]: Why partial class?

File: iam/api/Access/TestPermissions.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+{
+    public static IList<String> TestPermissions(string projectId)
```
Comment (simonz130) [inline iam/api/Access/TestPermissions.cs]: Can you rename to TestIamPermissions?

File: iam/api/AccessTest/AccessTest.cs [hunk 3]
```diff
@@ -36,19 +36,21 @@ public AccessTest()
...
             {
                 HttpClientInitializer = credential
             });
 
-            // Create custom roles for testing
-            var role1 = new Role
+            return service;
+        }
+
+        public Role createCustomRole(IamService service)
```
Comment (simonz130) [inline iam/api/AccessTest/AccessTest.cs]: public methods must start with capital letters in C#

File: iam/api/AccessTest/AccessTest.cs [hunk 4]
```diff
@@ -124,5 +123,11 @@ public void TestAccess()
                 service.Projects.Roles.Delete(role1.Name).Execute();
             }
         }
+
+        [Fact]
+        public void TestPermissions()
+        {
+            var permissions = AccessManager.TestPermissions(_project);
```
Comment (simonz130) [inline iam/api/AccessTest/AccessTest.cs]: You don't have any asserts here. What is the purpose of the test?

File: iam/api/Access/TestPermissions.cs (hunk 1 above)
Comment (melaniedejong) [reply iam/api/Access/TestPermissions.cs]: I decided to use a partial class to mirror the other Access snippets (i.e. AddBinding.cs). Please let me know if I should change it.

File: iam/api/AccessTest/AccessTest.cs (hunk 4 above)
Comment (melaniedejong) [reply iam/api/AccessTest/AccessTest.cs]: I'm expecting TestPermissions to throw an error if it fails (following the pattern of [QuickStartTest.cs](https://github.com/melaniedejong/dotnet-docs-samples/blob/master/iam/api/QuickStartTest/QuickStartTest.cs)). I'm not entirely sure what to assert--is there a standard for C# tests?

File: iam/api/Access/TestPermissions.cs (hunk 2 above)
Comment (melaniedejong) [reply iam/api/Access/TestPermissions.cs]: Done

File: iam/api/AccessTest/AccessTest.cs (hunk 3 above)
Comment (melaniedejong) [reply iam/api/AccessTest/AccessTest.cs]: Done

Comment (melaniedejong) [reply iam/api/AccessTest/AccessTest.cs]: Done

**Comment by melaniedejong** [comment]
The "maximum number of roles reached" error is back: 

Error Message:
   Google.GoogleApiException : Google.Apis.Requests.RequestError
Maximum number of roles reached. Maximum is: 300 [429]
Errors [
        Message[Maximum number of roles reached. Maximum is: 300] Location[ - ] Reason[rateLimitExceeded] Domain[global]
]

File: iam/api/AccessTest/AccessTest.cs (hunk 4 above)
Comment (SurferJeffAtGoogle) [reply iam/api/AccessTest/AccessTest.cs]: Nothing to assert.  Your test is fine.

**Comment by SurferJeffAtGoogle** [comment]
LGTM.  I'll merge this afternoon if no one objects.

**Pull request merged**